### DISCUSSION TITLE
Show list of currently running workflows under an account/organization

### DISCUSSION THREAD

**Discussion by victorlin** [community/community#50156] [Enterprise] [answered]
### Select Topic Area

Question

### Body

Due to the [account-level limits](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits) on the number of concurrent jobs for GitHub hosted runners, it would be useful to have an account-level overview of currently running jobs (or workflows) to understand real-time usage towards those limits.

I'm aware there are ways to show a list of currently running workflows on a specific repository (e.g. `https://github.com/<org>/<repo>/actions?query=is%3Ain_progress`), but this is not very useful if workflows are running under several repositories.

**Comment by Anmol-Baranwal** [comment]
Currently, GitHub does not provide an account-level overview of currently running jobs or workflows. However, there are a few workarounds that you could use to get this information:

1. Use the GitHub API: You can use the GitHub API to programmatically retrieve information about your running workflows. Specifically, you can use the List workflow runs for a repository endpoint (https://docs.github.com/en/rest/reference/actions#list-workflow-runs-for-a-repository) to retrieve a list of currently running workflows across all your repositories. You can then filter the results based on your needs.

2. Use a third-party tool: There are several third-party tools that can provide you with an overview of your GitHub workflows, including currently running workflows. One popular tool is CircleCI (https://circleci.com/integrations/github/), which provides a dashboard that shows real-time information about your workflows.

3. Monitor your workflows manually: If you only have a few workflows running across your repositories, you can monitor them manually by visiting each repository's Actions page and looking for workflows that are in progress.

While these workarounds may not provide a complete account-level overview of your running workflows, they can help you get a better understanding of your current usage towards account-level limits.

**Reply by Anmol-Baranwal** [reply]
@victorlin 
I just wanted to check in and see if you had any further questions or concerns about my answer on Github. If you have no further doubts or if your doubts have been resolved, would you mind marking my answer as accepted?

It would be greatly appreciated and it will also help other users who might be looking for a solution to a similar issue.

**Reply by victorlin** [reply]
@Anmol-Baranwal, thank you for your response. Sadly, none of the workarounds are acceptable to me:

1. I am looking for an official feature that can be used by anyone, not a custom script.
2. I peeked at the CircleCI dashboard, but it looks like repos must be added one-by-one, which also involves creating a CircleCI pipeline per repo. I don't want that, so I didn't try further.
3. This is what I do now and is impractical for accounts with many repositories.

**Comment by JakeWharton** [comment]
We recently had a repository accidentally create 2000 jobs which exhausted our enterprise concurrency limit (500) and prevented actions from working across 10+ orgs. It took five hours to figure out the root cause and find the offending repo (support got back to us with the offender at the same time we discovered it ourselves going one-by-one).

1. No single project should be able to run that many jobs at once. Having an organization level setting which limits per-project parallelism of jobs would have prevented this. For example, it seems perfectly reasonable to say that no repo across all of our orgs should ever get more than 10 running jobs at once. This ensures other repos can always make forward progress even in the face of one repo creating 2000 jobs.

2. It should be easier to locate the source of such an onslaught of jobs at the organization level. The actions tab view within a repo would be perfect at the organization level with an additional column for the project.

**Comment by victorlin** [comment]
I stumbled across @chriskinsman's [GitHub Action Dashboard](https://github.com/chriskinsman/github-action-dashboard) which looks promising, but requires some permissions setup and running via Docker so I haven't tried it yet.

For unofficial solutions, a tool like this is probably as good as it gets.

**Reply by glyph** [reply]
This is now unmaintained.

**Reply by chriskinsman** [reply]
Anyone who wants to take it over let me know.  Just not using anymore with a move to monorepo...

**Comment by victorlin** [comment, marked as answer]
This is now available in organization settings > Runners > GitHub-hosted runners. URL looks like `https://github.com/organizations/<name>/settings/actions/hosted-runners`. Screenshot:

![GitHub-hosted runners Active jobs](https://github.com/community/community/assets/13424970/cec544a1-c2fc-4765-b04e-ea68ca27f85a)

**Reply by jsoref** [reply]
It doesn't work for accounts, only orgs. Basically no user in their right mind would do any work in their own account because if they do they get no observability.

**Reply by mstorsjo** [reply]
Bump; it would be very useful to have the same view for regular accounts too.

**Reply by gian1200** [reply]
is there a similar view for all "queued" actions/workflows on an org?

**Reply by jsoref** [reply]
I don't think so

**Comment by imTachu** [comment]
If you don't have admin permissions over the org and just looking to find the actual actions code across repos, an org-wide search on Github does the trick: 

```
org:nameOfYourOrg path:.github/workflows/*.yml
```
